lib/active_merchant/billing/gateways/plexo.rb

```diff
@@ -285,10 +292,13 @@ def reorder_amount_fields(response)
       end
 
       def commit(action, parameters, options = {})
-        base_url = (test? ? test_url : live_url)
-        url = build_url(action, base_url)
-        response = parse(ssl_post(url, parameters.to_json, header(options)))
-        response = reorder_amount_fields(response) if AMOUNT_IN_RESPONSE.include?(action)
+        base_url = test? ? test_url : live_url
+        url = build_url(action, base_url, options, parameters)
+
+        raw = make_request(action, url, parameters, options)
```

Comment:
💬 Comment:

Maybe we could be more conservative here and follow a pattern (that's actually good) by changing `ssl_post` to `ssl_resquest` and setting another optional parameter on `commit` and setting 'post' as the default value, then in the inquiry that is the only one that deviates from the standard post send the optional parameter to specify a `get`

```ruby
      def commit(action, parameters, options = {}, method = :post)
        base_url = test? ? test_url : live_url
        url = build_url(action, base_url, options, parameters)

        response = parse(ssl_request(url, method, parameters.to_json, header(options)))
       
        # more code
        # ...
```

Reply:
yes, you are right however I am not sure what is the pattern since I can see several gateways using ssl_post, and this feature was based on the previous implementation for d_local gateway